sources/engine/Stride.Rendering/Rendering/ProceduralModels/PrimitiveProceduralModelBase.cs

```diff
@@ -51,6 +51,16 @@ public void SetMaterial(string name, Material material)
         [DataMember(530)]
         public Vector3 LocalOffset { get; set; }
 
+        /// <summary>
+        /// Gets or sets the number of texture coordinates to generate. A value between 1 and 10, inclusive.
+        /// </summary>
+        /// <value>
+        /// The number of texure coordinates.
+        /// </value>
+        [DataMember(540)]
+        [DataMemberRange(1, 10)]
+        public int NumOfTexCoords { get; set; } = 10;
```

Comment:
For API consistency, please use `NumberOfTexCoords`. Or maybe even `NumberOfTextureCoordinates` (it would matches `VertexElement.TextureCoordinate` and display better in property grid).

Note: if you want a more verbose version in property grid, feel free to add something like `[Display("Number of texture coordinate channels")]` or any description that you think is better if default one is not good enough.

Reply:
Sounds good, will improve on that. Which help text will be displayed in the help box below the property grid?

Reply:
@tebjan for the detail in the property grid lower help area, the proper tag is `<userdoc>Some description</userdoc>` (you can put it after `</value>`)